pkg/secrets/keyvault/client.go

```diff
@@ -52,23 +60,29 @@ func getVaultsURL(ctx context.Context, vaultName string) string {
 // redundant since that's in the credentials, but it's used to
 // override the one specified in credentials so it might be right to
 // keep it. Confirm this.
-func New(keyvaultName string, creds config.Credentials) *KeyvaultSecretClient {
+func New(keyVaultName string, creds config.Credentials, secretNamingVersion secrets.SecretNamingVersion) *SecretClient {
 	keyvaultClient := keyvaults.New()
 	a, _ := iam.GetKeyvaultAuthorizer(creds)
 	keyvaultClient.Authorizer = a
 	keyvaultClient.AddToUserAgent(config.UserAgent())
-	return &KeyvaultSecretClient{
-		KeyVaultClient: keyvaultClient,
-		KeyVaultName:   keyvaultName,
+	return &SecretClient{
+		KeyVaultClient:      keyvaultClient,
+		KeyVaultName:        keyVaultName,
+		SecretNamingVersion: secretNamingVersion,
 	}
-
 }
 
-func IsKeyVaultAccessible(kvsecretclient secrets.SecretClient) bool {
+// TODO: This method is awkward -- move to interface?
+func IsKeyVaultAccessible(client secrets.SecretClient) bool {
 	ctx := context.Background()
-	key := types.NamespacedName{Name: "", Namespace: ""}
+	key := secrets.SecretKey{Name: "test", Namespace: "default", Kind: "test"}
+
+	data, err := client.Get(ctx, key)
+	if err == nil {
+		// TODO: It's not expected we ever take this path -- should this be an error?
+		return false
+	}
```

Comment:
Comparing `err == nil` is jarring. Is that correct?

Reply:
It _is_ correct but there's a clearer way to write this method that also has the advantage of working if we previously had leaked the dummy key we're using to test.

Really this method needs to go away entirely (what it's doing is really not good), but I've kept it here for now because I didn't want to remove functionality that I don't fully understand why we have.

Reply:
The reworked method looks clearer. 👍